segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/ThrottlerCalculator.java

```diff
@@ -136,32 +145,49 @@ DelayResult getThrottlingDelay() {
          *
          * @return
          */
-        abstract String getName();
+        abstract ThrottlerName getName();
     }
 
     /**
      * Calculates the amount of time to wait before processing more operations from the queue in order to relieve pressure
      * on the cache. This is based on ReadIndex statistics, mainly cache utilization ratio.
      */
-    @RequiredArgsConstructor
     private static class CacheThrottler extends Throttler {
+        private final double targetCacheUtilization;
+        private final int baseDelay;
+        @NonNull
         private final Supplier<Double> getCacheUtilization;
 
+        CacheThrottler(Supplier<Double> getCacheUtilization, double targetCacheUtilization, double maxCacheUtilization) {
+            this.targetCacheUtilization = targetCacheUtilization + CACHE_TARGET_UTILIZATION_THRESHOLD_ADJUSTMENT;
+            this.getCacheUtilization = getCacheUtilization;
+            if (this.targetCacheUtilization >= maxCacheUtilization) {
+                // Since this is externally provided, we need to be able to handle invalid values. If we get too small of
+                // a utilization, then we will apply maximum throttle as soon as we reach the min utilization threshold.
+                this.baseDelay = MAX_DELAY_MILLIS;
+            } else {
+                this.baseDelay = calculateBaseDelay(maxCacheUtilization, this::getDelayMultiplier);
```

Comment:
Am I reading right that this `baseDelay` is `25000 / getDelayMultiplier(maxCacheUtilization)`? If so and the max cache utilization is 0.85, then this is:

```
25000 / (100 * ( utilization - 0.85 )
```

Say that utilization is `0.90` for the sake of argument. The `baseDelay` in this case is 5s. If the utilization goes up to say `0.95`, then the base delay becomes 2.5s. Either I'm not reading this right or the fact that the base delay goes down with the difference is counterintuitive to me. Could you clarify?

Reply:
No. This is a static value calculated once per process, with input from the configuration. The configuration will not change.

If target = 0.90 and max = 0.98, then this will be 25000/(100*0.13)=1.92s. The throttler will increase the delay by 1.92 seconds for every utilization percentage point above 90%.

If target = 0.95 and max = 0.98, then this will be 25000/(100*0.03)=8.33s. The throttler will increase the delay by 8.33 seconds for every utilization percentage point about 0.95. This is a steeper climb than the above since your target-max window is so narrow.

Reply:
I think you meant to say `0.85` in the first one, but I get the point in any case, a wider gap warrants a smaller base delay. Thanks for the clarification.